ddl/ddl_worker.go

```diff
@@ -175,14 +190,18 @@ func (d *ddl) addDDLJob(ctx sessionctx.Context, job *model.Job) error {
 	job.Version = currentVersion
 	job.Query, _ = ctx.Value(sessionctx.QueryString).(string)
 	err := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
-		t := meta.NewMeta(txn)
+		t := newMetaWithQueueTp(txn, job.Type.String())
```

Comment:
Why not use meta.AddIndexJobListKey as the second parameter?

Reply:
Almost. I want to put the check of job type or worker type into `newMetaWithQueueTp`.